17-lifecycle-lab/src/03-custominput.jsx

```diff
@@ -1,10 +1,59 @@
-/**
- * Implment an input component that shows 4
- * input boxes writing user input in the currently
- * focused panel
- *
- * See lab description on the webpage for a live example
- * how this should work.
- */
-
-
+import React from 'react';
+
+export default class CustomInput extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {};
+
+        this.elementsNumber = 4;
+    }
+
+    onKeyPressEvent = (e) => {
+        // get next element to be focused. return to the first item when it's reach the end.
+        let rec = document.getElementById(((Number(e.target.id) + 1) % this.elementsNumber));
+        rec.focus();
+    };
+
+    render() {
+        return (<div>
+            {[...Array(this.elementsNumber)].map((element, key) => (
+                <FocusedElement key={key} keyValue={key} onKeyPressEvent={this.onKeyPressEvent}/>
+            ))}
+        </div>);
+    }
+}
+
+class FocusedElement extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            text: "",
+        }
+    }
+
+    static PropTypes = {
+        keyValue: React.PropTypes.number.isRequired,
+        onKeyPressEvent: React.PropTypes.func.isRequired,
+    };
+
+    onKeyPressEvent = (e) => {

```

Comment:
change the name. It's confusing since you have the same function passed via props and on the class itself.